addons/stock/models/stock_location.py

```diff
@@ -98,7 +98,8 @@ def default_get(self, fields):
     forecast_weight = fields.Float('Forecasted Weight', compute="_compute_weight")
 
     _sql_constraints = [('barcode_company_uniq', 'unique (barcode,company_id)', 'The barcode for a location must be unique per company !'),
-                        ('inventory_freq_nonneg', 'check(cyclic_inventory_frequency >= 0)', 'The inventory frequency (days) for a location must be non-negative')]
+                        ('inventory_freq_nonneg', 'check(cyclic_inventory_frequency >= 0)', 'The inventory frequency (days) for a location must be non-negative'),
+                        ('replenish_location_has_warehouse', 'check(warehouse_id IS NOT NULL OR NOT replenish_location)', 'A location cannot be set as replenish location if it does not belong to any warehouse')]
```

Comment:
I think that we used to need to add python @api.constrains in stable instead of adding sql constraint (I think because if some data in customer database is wrong, we don't want to have an error when they update the module. Maybe it's ok now.